src/sink/kafka.rs

```diff
@@ -245,7 +246,8 @@ impl Sink for Kafka {
         for (value, meta) in event.value_meta_iter() {
             let encoded = codec.encode(value)?;
             let processed = postprocess(self.postprocessors.as_mut_slice(), ingest_ns, encoded)?;
-            let meta_kafka_key = meta.get_str("kafka_key");
+            let meta_kafka_key = meta.get("kafka_key").and_then(Value::as_str);
+            let meta_kafka_headers = meta.get("kafka_headers").and_then(Value::as_object);
```

Comment:
and a `get_object` i think :)

Reply:
I found that for objects, `get` followed by `unwrap_or_default()` is the easiest way, preserving the `value` allows to use `get_<type>` afterward and if the result is `default` it's `null` and all successive calls to `get_` will return `None`

Reply:
I don't think we can use `unwrap_or_default()` as `get` returns a `Value`, and I don't think any structs implementing `Value` have `Default` as derivable. We could opt for using `unwrap_or` and return `None`, if there is no `Some` value though. I'm still fairly new to the codebase, so please correct me if I'm wrong here :)

Reply:
If Default isn't implemented that would be an oversight on my part :) Value should default to 'null' that might have slipped away in some copy & pasta. Sorry if it's missing